### PR TITLE
Attack parameters overhaul

### PR DESCRIPTION
**Pull request by agjak**
**Main changes:**
- Added the "FoolboxParameters" and "ARTParameters" classes, both descended from a "Parameters" class
- Changed all attacks and tests, so that they all use the classes mentioned

**Additional changes:**
- Changed Brendel-Bethge attack, so that it fits the implementation of other Foolbox attacks
- Modified some of the default parameters in Foolbox tests to make the tests give better results.
- Added the verify_bounds function from Brendel Bethge to other foolbox attacks, since it is a good idea. This function automatically sets min and max values if they are not provided, based on the smallest and the biggest value included in the input provided.
- Added the verify_epsilon function to foolbox attacks, which automatically sets epsilon value if it isn't provided; the default value is (max-min)*0.001
- Refactored foolbox tests to use unittest, so that all tests can be ran at once.

### CODE REVIEW

File: attacks/foolbox_attack.py [hunk 1]
```diff
@@ -5,17 +5,36 @@
...
+        self.min = input_values.min().item()
+        self.max = input_values.max().item()
+
+    def verify_epsilon(self):
+        if self.epsilon is not None:
+            return
+        elif self.epsilon_rate is not None:
+            self.epsilon = (self.max - self.min)*self.epsilon_rate
+        else:
+            self.epsilon = (self.max-self.min)*0.001
```
Comment (makowskiignacy) [inline attacks/foolbox_attack.py:36]: to brzmi sensowniej

File: tests/test_frameworks.py
Comment (makowskiignacy) [inline tests/test_frameworks.py:1]: Nie jestem przekonany do usnięcia timeoutów, ale mogę dać się przekonać

Comment (agjak) [reply tests/test_frameworks.py:1]: Timeouty nie działają na windowsie, i wszystkie testy z nimi mi się wywalają, musiałam je usunąć żeby móc je puścić

File: attacks/art_attack.py
[the remark this answers is not in the document]
Comment (agjak) [reply attacks/art_attack.py:1]: Nie wiedziałam o tym, pousuwam

File: tests/test_brendel_bethge.py
[the remark this answers is not in the document]
Comment (agjak) [reply tests/test_brendel_bethge.py:1]: Dobry pomysł, zrobię to

File: tests/test_basic_iterative.py
[the remark this answers is not in the document]
Comment (agjak) [reply tests/test_basic_iterative.py:1]: Dobry pomysł, zrobię to

**Pull request merged**